shapeshift/web
# fix: defi allowance checks

## Description

This updates the checks across all DeFi sections providers (including the FOX Farming / LP ones) to use `allowance.gte(amount)` checks vs `allowance.gt(amount)` currently, both on poll and on initial allowance checks. 

## Notice




## Pull Request Type


## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/2461

## Risk

None, also tested as a sanity check in Fox Farming where exact allowance isn't implemented yet and this also works.

## Testing

- Start the staking flow in any DeFi opportunity, with an amount greater than your current allowance
- After `approve()` ing the contract to spend your tokens (can be either an exact or infinite allowance), the modal flow should continue to the "Confirm" step - don't confirm it yet, see next step

- Reload the page and try inputting the same amount of tokens in that opportunity
- The modal should skip the Approve step and go directly to the Confirm one

## Screenshots (if applicable)
